src/main/java/com/sahabuddin/blogappapis/services/impl/UserServiceImpl.java

```diff
@@ -60,6 +75,23 @@ public void deleteUser(Long userId) {
         this.userRepository.delete(user);
     }
 
+    @Override
+    public String verify(SignInRequest request) {
+        log.info("Verifying user: {}", request);
+        log.info(userDetailsService.loadUserByUsername(request.getUsername()).getUsername());
+        UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword());
+        try{
+            log.info("Attempting to verify user {}", usernamePasswordAuthenticationToken);
+            authenticationManager.authenticate(usernamePasswordAuthenticationToken);
+            log.info("Authentication Successful");
+            UserDetails userDetails = userDetailsService.loadUserByUsername(request.getUsername());
+            return jwtTokenHelper.generateToken(userDetails);
+        } catch (BadCredentialsException e) {
+            log.error(e.getMessage());
+            return "fail";
```

Comment:
This will return a String "fail" if fails to generate a JWT which is not correct. Throw a custom exception with a meaningful message if it fails to generate JWT.